### DISCUSSION TITLE
profiles

### DISCUSSION THREAD

**Discussion by mr-cheffy** [zen-browser/desktop#805] [Ideas]
### What feature would you like to see?

hello!
this is a feature that as far as i'm aware only arc browser has, but is the entire reason i use the broswer,
it's the concept of "profiles" where if i switch from one profile to another it's effectively another browser in the same window; with all my client info, accounts, cookies, etcetc being completely wiped and being an entirely new instance.

### What platform are you seeing the problem on?

macOS

### Additional context

this is a killer feature for people like me who attempt to keep their personal life and professional life separate and allows me to easily separate sensitive information such as bank logins and what not into one profile, then if i'm saying doing something like streaming then i can have a seperate profile where everything is logged out and i don't have to worry about any crossing of information/sensitivity.
 
 ~ _Created automatically from issue: https://github.com/zen-browser/desktop/issues/532_

**Comment by shanx** [comment]
Yeah I second this! For me a specific use case of this is that I have a lot of the same SaaS platforms that I use in professional projects, that I want each workspace to have an open session for different identity. So that I don't have to login / log out or have multiple identities logged in at once.

**Comment by sjelfull** [comment]
Yes please! This separation between work and personal is a killer feature, and only Arc has nailed it. Other browsers are just so bad at this, I don't even bother and live with the fact that i'm suddenly saving something on Youtube to my work account, or whatever

**Comment by Shape6253** [comment]
This function is already implemented. You just need to assign a separate container to the Workspace. Click on the three dots next to the Workspace name, select "Set Default Container," and choose the desired one. You can create a new container in the browser settings.

**Reply by Diraq-Girach** [reply]
"Currently Container Tabs separate your cookies/browsing sessions, but it doesn't separate your browsing history and extensions." and as far as I could find, the passwords are also common 

I can be wrong though. But a completely isolated environment like arc would be great.

**Comment by shanx** [comment]
@Shape6253 amazing, you're absolutely right! It might be nice to make this clearer to people.

**Comment by Shape6253** [comment]
One possible solution is to provide the option to automatically create a new container when setting up a workspace. Users should also see a clear explanation so they understand that all sites within it are isolated from others by default. Otherwise, users will start asking not how to isolate, but why they are isolated

**Reply by shanx** [reply]
In Arc I was used to configure this explicitly myself also. But I'm used to this feature being called a Profile instead of `Container tabs`. I see that in the world of Firefox/Zen this is a specific invention Mozilla made to tackle this same problem. I just didn't expect this naming, but that is me coming from Chrome based browsers. To somehow mention this would already be a big help.

And then yes I think it would really help that when you create a new workspace it explicitly allows you to create it as an isolated workspace. Which essentially means auto creating a Container for this workspace and using it for that workspace. By default that option should be switched off.

**Comment by raymondbh** [comment]
Containers does not handle different browser settings, ex. proxy settings and more... So a easier way to chose profile is very welcome!

**Comment by edjw** [comment]
Isolating addons to specific workspaces is not possible with containers as far as I can tell. I think this is only doable in Firefox with profiles. For example. I'd like to be able to only use my personal password manager in my Personal Zen workspace and the same for my work password manager and Work workspace

**Comment by Dulanjala007** [comment]
when searching history you can see every profile's history at once...
It's not good for privacy
it's nightmare to look up browsing history to find a certain profile's urls quickly, in the order you visited them

**Reply by rsenna** [reply]
Yeah, agreed. But it's pretty much a nightmare _inherited from Firefox_.
There is so much Zen developers can do.
In this particular case, I suspect they would have to _stop using Firefox Multi-Account Containers and start fresh_. That would take time…
